src/Renci.SshNet/Renci.SshNet.csproj

```diff
@@ -5,19 +5,23 @@
 	<TargetFrameworks>net462;netstandard2.0;netstandard2.1;net6.0;net7.0</TargetFrameworks>
   </PropertyGroup>
 
+  <PropertyGroup Condition=" $([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0')) ">
+    <IsTrimmable>true</IsTrimmable>
+  </PropertyGroup>
```

Comment:
I don't think we need to enable this feature directly in the library, right?

Reply:
According to https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/prepare-libraries-for-trimming (if I have understood correctly) is it a library-level flag to declare that the library is compatible with trimming. It also enables the analysers.

So I think we either need `IsTrimmable` or `EnableTrimAnalyzer`, and because this PR fixes all the trimming warnings, then we can have `IsTrimmable`.

Reply:
This is good answer :)

Reply:
@Rob-Hague, do you happen to know how exactly it "marks the assembly as "trimmable"? Do we also mark **SshNet.Security.Cryptography** as trimmable? I would've expect a warning because we have a dependency that is marked trimmable.

Reply:
Good point, no, I don't know how it works with references. I have changed it just to `EnableTrimAnalyzer` until someone more knowledgeable comes along.